optimum/onnxruntime/modeling_ort.py

```diff
@@ -356,23 +361,27 @@ def _from_transformers(
                 kwargs will be passed to the model during initialization
         """
         # create local save dir in cache dir
-        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir = Path(save_dir).joinpath(model_id, subfolder)
         save_dir.mkdir(parents=True, exist_ok=True)
         kwargs["model_save_dir"] = save_dir
 
         # reads pipeline task from ORTModelForXXX class if available else tries to extract from hub
         if cls.export_feature is not None:
             task = cls.export_feature
         else:
-            task = HfApi().model_info(model_id, revision=revision).pipeline_tag
+            task = HfApi().model_info(model_id, revision=revision).pipeline_tag  # FIXME: load from subfolder?
```

Comment:
Out of curiosity, the FIXME comment means that we are not sure if this can be done with `HfApi`?

Reply:
One of the issue is that the ONNX export in transformers relies on preprocessors to generate dummy inputs, hence we use `preprocessor = get_preprocessor(model_id)` that has no option to specify a subfolder. We would need to do a PR in transformers for that. Therefore, the tokenizer must be at the top level of the repo, as well as the `config.json` that may be required in `AutoTokenizer` to find the right class.

I changed a bit the code so that we don't need to expect a `config.json` in the subfolder, we fall back to using the one at the top level if there is not in the subfolder. Does it sound good to you?

Reply:
Yes sounds good! In that case, maybe we could log a warning saying that the config was not found in the given subfolder and that we are looking for it in the parent folder.